pkg/sql/logictest/logic.go

```diff
@@ -1170,19 +1173,27 @@ func (t *logicTest) outf(format string, args ...interface{}) {
 }
 
 // setSessionUser sets the DB client to the specified user and connects
-// to the node in the cluster at index nodeIdx.
-func (t *logicTest) setSessionUser(user string, nodeIdx int) {
-	db := t.getOrOpenClient(user, nodeIdx)
+// to the node in the cluster at index nodeIdx. If forceNew is specified,
+// the current client for the (user, nodeIdx) combination (if one exists)
+// will be closed and a new one will be created.
+func (t *logicTest) setSessionUser(user string, nodeIdx int, forceNew bool) {
+	db := t.getOrOpenClient(user, nodeIdx, forceNew)
 	t.db = db
 	t.user = user
 	t.nodeIdx = nodeIdx
 }
 
 // getOrOpenClient returns the existing client for the given user and nodeIdx,
-// if one exists. Otherwise, it opens and returns a new client.
-func (t *logicTest) getOrOpenClient(user string, nodeIdx int) *gosql.DB {
+// if one exists. Otherwise, it opens and returns a new client. If forceNew is
+// specified, the existing client (if exists) will be closed and a new one will
+// be opened.
+func (t *logicTest) getOrOpenClient(user string, nodeIdx int, forceNew bool) *gosql.DB {
```

Comment:
Do you think there'd be more future utility out of this functionality if you could instead of terminating and reopening a connection you exposed a session handle or name? Tests could then specify `user <name> session=notthedefault` to establish a new connection. Or is the important aspect terminating the active session rather than creating a new one?

Reply:
Another PR I currently have open (that depends on this one) is to add a regression test for the temp view descriptor persistence bug we recently discovered: https://github.com/cockroachdb/cockroach/pull/109112. For that one, it is important that we end the previous session because we want to verify that the temp view's descriptor is dropped and this only happens after the session ends.